shared.ts

```diff
@@ -1,82 +1,75 @@
 // Internal shared types
 import type {
   ApplicationCommandOption,
-  ApplicationCommandOptionTypes,
   ApplicationCommandTypes,
   Bot,
   Interaction,
 } from "./deps.ts";
 
+import type { InteractionDataOption, InteractionResponse } from "./deps.ts";
+
 import type { Loggable } from "./src/logger/mod.ts";
 import type { Config } from "./src/config/mod.ts";
 
-import { CommandManager } from "./src/command/commander.ts";
+export type { Config, Loggable };
 
-export type { Loggable };
+export interface CommandRegistry {
+  init: (bot: Bot, config: Config) => void;
+  getCommand: (interaction: Interaction) => Command | Error;
+
+  registerGuildCommands: (
+    bot: Bot,
+    guildIds: bigint[],
+  ) => Promise<Error | void>;
+}
 
 export interface HandlerOptions {
   logger: Loggable;
   config: Config;
-  cmdMgr: CommandManager;
+  cmdMgr: CommandRegistry;
 }
 
-export type CommandHandler = (bot: Bot, interaction: Interaction) => unknown;
+export interface Argument {
+  name: string;
+  type: InteractionDataOption;
+  value: any;
+}
+
+export type ExecResult = InteractionResponse | void | Error;
+
+export interface Executor {
+  beforeExec(): Promise<ExecResult>;
+  execute(): Promise<ExecResult>;
+  afterExec(): Promise<ExecResult>;
+}
 
 export interface Command {
```

Comment:
Command represents a slash command on discord.
SubCommand represents a its sub command.

It is the role of the Executor to actually process those slash commands.